python/feathub/common/types.py

```diff
@@ -38,7 +42,10 @@ class BasicDType(Enum):
 
 class DType(ABC):
     def __init__(self) -> None:
-        pass
+        self.to_json = append_metadata_to_json(  # type: ignore
```

Comment:
It is not easy to know whether `to_json` is decorated here by reading the code.

To make the code more readable, can we either use the python decorator by annotating the `to_json()` with `@append_metadata_to_json()`, or add the metadata (e.g. version, type) directly in the original `to_json()`?

Note that we can use a constant global variable to specify the current json format version.